Minecraft/src/main/java/com/microsoft/Malmo/MissionHandlers/DiscreteMovementCommandsImplementation.java

```diff
@@ -39,6 +52,98 @@
     private boolean isOverriding;
     private int direction = -1;
 
+    public static class UseActionMessage implements IMessage
+    {
+        String parameters;
+        public UseActionMessage()
+        {
+        }
+    
+        public UseActionMessage(String parameters)
+        {
+            this.parameters = parameters;
+        }
+
+        @Override
+        public void fromBytes(ByteBuf buf)
+        {
+            this.parameters = ByteBufUtils.readUTF8String(buf);
+        }
+
+        @Override
+        public void toBytes(ByteBuf buf)
+        {
+            ByteBufUtils.writeUTF8String(buf, this.parameters);
+        }
+    }
+
+    public static class UseActionMessageHandler implements IMessageHandler<UseActionMessage, IMessage>
+    {
+        @Override
+        public IMessage onMessage(UseActionMessage message, MessageContext ctx)
+        {
+            MovingObjectPosition mop = Minecraft.getMinecraft().objectMouseOver;
+            if( mop.typeOfHit == MovingObjectPosition.MovingObjectType.BLOCK )
+            {
+                EntityPlayerMP player = ctx.getServerHandler().playerEntity;
+                if( player.getCurrentEquippedItem() != null ) {
+                    ItemStack itemStack = player.getCurrentEquippedItem();
+                    Block b = Block.getBlockFromItem( itemStack.getItem() );
+                    if( b != null ) {
+                        BlockPos pos = mop.getBlockPos().add( mop.sideHit.getDirectionVec() );
+                        IBlockState blockType = b.getStateFromMeta( itemStack.getMetadata() );
+                        MinecraftServer.getServer().getEntityWorld().setBlockState( pos, blockType );
```

Comment:
Still not entirely sure that MinecraftServer.getServer() is safe. `ctx.getServerHandler().playerEntity.worldObj` should be safe.